tests/unit/python/foglamp/services/core/test_server.py

```diff
@@ -0,0 +1,435 @@
+        request_data = {"type": "Storage", "name": "Storage Services", "address": "127.0.0.1", "service_port": 8090, "management_port": 1090}
+        with patch.object(ServiceRegistry, 'register', return_value=None) as patch_register:
+            resp = await client.post('/foglamp/service', data=json.dumps(request_data))
+            assert 400 == resp.status
+            assert 'Service {} could not be registered'.format(request_data['name']) == resp.reason
+        args, kwargs = patch_register.call_args
+        assert (request_data['name'], request_data['type'], request_data['address'],  request_data['service_port'], request_data['management_port'], 'http') == args
+
+    async def test_register_service(self, client):
+        async def async_mock(return_value):
+            return return_value
+
+        request_data = {"type": "Storage", "name": "Storage Services", "address": "127.0.0.1", "service_port": 8090, "management_port": 1090}
+        with patch.object(ServiceRegistry, 'register', return_value='1') as patch_register:
+            with patch.object(AuditLogger, '__init__', return_value=None):
+                with patch.object(AuditLogger, 'information', return_value=async_mock(None)) as audit_info_patch:
+                    resp = await client.post('/foglamp/service', data=json.dumps(request_data))
+                    assert 200 == resp.status
+                    r = await resp.text()
+                    json_response = json.loads(r)
+                    assert {'message': 'Service registered successfully', 'id': '1'} == json_response
+                args, kwargs = audit_info_patch.call_args
+                assert 'SRVRG' == args[0]
+                assert {'name': request_data['name']} == args[1]
+        args, kwargs = patch_register.call_args
+        assert (request_data['name'], request_data['type'], request_data['address'], request_data['service_port'], request_data['management_port'], 'http') == args
+
+    async def test_service_not_found_when_unregister(self, client):
+        with patch.object(ServiceRegistry, 'get', side_effect=service_registry_exceptions.DoesNotExist) as patch_unregister:
+            resp = await client.delete('/foglamp/service/blah')
+            assert 404 == resp.status
+            assert 'Service with blah does not exist' == resp.reason
+        args, kwargs = patch_unregister.call_args
+        assert {'idx': 'blah'} == kwargs
+
+    async def test_unregister_service(self, client):
+        async def async_mock():
+            return ""
+
+        service_id = "c6bbf3c8-f43c-4b0f-ac48-f597f510da0b"
+        sname = "name"
+        stype = "Southbound"
+        sprotocol = "http"
+        saddress = "localhost"
+        sport = 1234
+        smgtport = 4321
+        data = []
+        record = ServiceRecord(service_id, sname, stype, sprotocol, saddress, sport, smgtport)
+        data.append(record)
+        Server._storage_client = MagicMock(StorageClient)
```

Comment:
as its used now again, can not we have a resuable def  to return service_records?